tests/models/test_hooks.py

```diff
@@ -500,6 +500,54 @@ def test_trainer_model_hook_system_test(tmpdir):
     assert called == expected
 
 
+def test_trainer_model_hook_system_fit_resume(tmpdir):
+    # initial training to get a checkpoint
+    model = BoringModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_steps=1,
+        limit_val_batches=0,
+        progress_bar_refresh_rate=0,
+        weights_summary=None,
+    )
+    trainer.fit(model)
+    best_model_path = trainer.checkpoint_callback.best_model_path
+    # resume from checkpoint with HookedModel
+
+    called = []
+    model = HookedModel(called)
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_steps=1,
+        limit_val_batches=0,
+        progress_bar_refresh_rate=0,
+        weights_summary=None,
+        resume_from_checkpoint=best_model_path,
+    )
+    assert called == []
+    trainer.fit(model)
+    expected = [
+        'prepare_data',
+        'configure_callbacks',
+        'setup',
+        'on_load_checkpoint',
+        'configure_sharded_model',
+        'configure_optimizers',
```

Comment:
Hey @carmocca I added a test here using your hook system for the setting 
Trainer(resume_from_checkpoint=..) and this will call on_load_checkpoint() on the model, which I have asserted here to run after setup and before configure_sharded model, which is what we need for the linked issue we try to resolve. 

overall, does this test look ok to you?

Reply:
Yes.

IMO you could merge it with `test_trainer_model_hook_system_fit_no_val`.
Do you see a benefit in having them separate?

Reply:
Okay, I just need to save a checkpoint beforehand, in that test so we have something to load.

Reply:
made the merge, let me know if something is unclear. added comments